RV-Monitor/Kernel_Safety_Claims_by_Runtime_Verification_Monitors.md

```diff
@@ -0,0 +1,542 @@
+<!-----
+
+You have some errors, warnings, or alerts. If you are using reckless mode, turn it off to see inline alerts.
+* ERRORs: 0
+* WARNINGs: 0
+* ALERTS: 3
+
+Conversion time: 2.825 seconds.
+
+
+Using this Markdown file:
+
+1. Paste this output into your source file.
+2. See the notes and action items below regarding this conversion run.
+3. Check the rendered output (headings, lists, code blocks, tables) for proper
+   formatting and use a linkchecker before you publish this page.
+
+Conversion notes:
+
+* Docs to Markdown version 1.0β35
+* Mon Jan 15 2024 04:51:24 GMT-0800 (PST)
+* Source doc: Kernel Safety Claims by Runtime Verification Monitors
+* Tables are currently converted to HTML tables.
+* This document has images: check for >>>>>  gd2md-html alert:  inline image link in generated source and store images to your server. NOTE: Images in exported zip file from Google Docs may not appear in  the same order as they do in your doc. Please check the images!
+
+
+WARNING:
+You have 7 H1 headings. You may want to use the "H1 -> H2" option to demote all headings by one level.
+
+----->
+
+# Kernel Safety Claims by Runtime Verification Monitors
+
+
+# Introduction
+
+The goal of this document is to define a comprehensive methodology to design an RV Monitor against a set of allocated safety requirements. \
+Instead we do not intend to prove the RV Monitor to be able to accommodate every single use case.
+
+
+### Why it is hard to qualify the Kernel to support associated safety claims
+
+The Linux Kernel is monolithic and provides thousands of functionalities in order to support a huge amount of applications across multiple industry domains and across many different target HW platforms.
+
+For this reason the Kernel is designed to be modular (see [MAINTAINERS](https://github.com/torvalds/linux/blob/master/MAINTAINERS)), however each Kernel element (subsystem or driver) provides many functionalities and all such elements can interact with each other in a very complex way. To understand how complex such interactions can be, fig. [1] in the Appendix shows the possible call tree in the context of ioctl between “FILESYSTEMS (VFS and infrastructure)” and the drivers or subsystems it communicates with. \
+ \
+As an example, if there is a safety requirement associated with ioctl() (e.g. ioctl shall be used to set the correct timeout in an external safety watchdog device), even if the operation is in principle very simple, the Kernel code providing the implementation must be qualified to make sure:
+
+
+
+1. The kernel code implementing the requirement is correct;
+2. There is no unintended functionality in the current implementation;
```

Comment:
I would replace "implementation" with module(s) or subsystem(s)

Or perhaps "implementation of subsystems involved"

Same for the rest.

Reply:
I'll propose a rewording to disambiguate between the whole active Kernel code AND the actual Kernel code implementing the ioctl syscall